src/Disks/S3/registerDiskS3.cpp

```diff
@@ -49,9 +49,8 @@ void checkReadAccess(const String & disk_name, IDisk & disk)
 void checkRemoveAccess(IDisk & disk) { disk.removeFile("test_acl"); }
 
 std::shared_ptr<S3::ProxyResolverConfiguration> getProxyResolverConfiguration(
-    const String & prefix, const Poco::Util::AbstractConfiguration & proxy_resolver_config)
+    const String & prefix, const Poco::Util::AbstractConfiguration & proxy_resolver_config, const Poco::URI & endpoint)
 {
-    auto endpoint = Poco::URI(proxy_resolver_config.getString(prefix + ".endpoint"));
```

Comment:
It was intentional, see 
https://github.com/ClickHouse/ClickHouse/blob/6d5137879b6b3c95cf5ad6f6d9bd312a3e066199/tests/integration/test_s3_with_proxy/configs/config.d/storage_conf.xml#L25-L30